### ISSUE TITLE
NLog support for MA0124 and MA0135

### ISSUE THREAD

**Issue by Egoushka** [meziantou/Meziantou.Analyzer#876]
**Is your feature request related to a problem? Please describe.**

Currently, MA0124 and MA0135 do not correctly recognize NLog when it is used as the logging provider.
This leads to false positives or missing checks when NLog is used instead of Microsoft.Extensions.Logging.ILogger.

**Describe the solution you'd like**
It would be great if MA0124 and MA0135 could support NLog structured logging patterns in the same way they currently support Microsoft.Extensions.Logging and Serilog.

**Additional context**
Example of NLog usage that currently triggers the analyzer incorrectly:
```
private static readonly Logger Logger = LogManager.GetCurrentClassLogger();

public void Test()
{
    Logger.Info("User {UserId} logged in", userId);
}
```

**Comment by meziantou** [comment]
> This leads to false positives

Are you saying that the analyzer is reporting a diagnostic when it shouldn't? If so, could you provide a repro? I'm not able to reproduce the issue on my machine.

> This leads to missing checks when NLog is used instead of Microsoft.Extensions.Logging.ILogger

I don't have time to work on this at the moment, but if someone is willing to submit a PR, I'd be happy to review it.

**Comment by Egoushka** [comment]
> Are you saying that the analyzer is reporting a diagnostic when it shouldn't? If so, could you provide a repro? I'm not able to reproduce the issue on my machine.

I mean, it's not showing missing or incorrect log parameters when you're using NLog, that's what I mean by false positive.
